rust/azure_iot_operations_mqtt/src/session/session.rs

```diff
@@ -242,44 +245,91 @@ impl Session {
-                    //     match sat_auth_tx.send(auth.code) {
-                    //         Ok(()) => {}
-                    //         Err(e) => {
-                    //             // This should never happen unless the background task has exited
-                    //             // in which case the session is already in a bad state and we should
-                    //             // have already exited.
-                    //             log::error!("Error sending auth code to SAT auth task: {e:?}");
-                    //         }
-                    //     }
-                    // }
-                } else {
-                    // TODO: handle failure once that's implemented on mqtt client
-                    self.connect_handle
-                        .take()
-                        .unwrap()
-                        .connect(
-                            self.connect_parameters.connection_transport_config.clone(),
-                            clean_start,
-                            self.connect_parameters.keep_alive,
-                            self.connect_parameters.will,
-                            self.connect_parameters.username,
-                            self.connect_parameters.password,
-                            self.connect_parameters.connect_properties,
-                            Some(self.connect_parameters.connection_timeout),
-                        )
-                        .await
-                };
+            let (connection, connack, disconnect_handle) = if let Some(authentication_info) =
+                authentication_info
+            {
+                unimplemented!()
+                // self.connect_handle.connect_with_auth(
+                // log::debug!("Incoming AUTH: {auth:?}");
+                //
+                // if let Some(sat_auth_tx) = &sat_auth_tx {
+                //     // Notify the background task that the auth data has changed
+                //     // TODO: This is a bit of a hack, but it works for now. Ideally, the reauth
+                //     // method on rumqttc would return a completion token and we could use that
+                //     // in the background task to know when the reauth is complete.
+                //     match sat_auth_tx.send(auth.code) {
+                //         Ok(()) => {}
+                //         Err(e) => {
+                //             // This should never happen unless the background task has exited
+                //             // in which case the session is already in a bad state and we should
+                //             // have already exited.
+                //             log::error!("Error sending auth code to SAT auth task: {e:?}");
+                //         }
+                //     }
+                // }
+            } else {
+                match self
+                    .connect_handle
+                    .take()
+                    .unwrap()
```

Comment:
Leave a comment indicating this cannot fail because a handle will always be present when running this function

Reply:
added!